web_app_drawer/static/src/js/web_app_drawer.js

```diff
@@ -0,0 +1,232 @@
+/* Copyright 2016 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+
+odoo.define('web_app_drawer.AppDrawer', function(require) {
+    'use strict';
+    
+    var $ = require('$');
+    var Menu = require('web.Menu');
+    var Class = require('web.Class');
+    var SearchView = require('web.SearchView');
+    var core = require('web.core');
+    
+    Menu.include({
+        // Force all_outside to prevent app icons from going into more menu
+        reflow: function() {
+            this._super('all_outside');
+        },
+    });
+    
+    SearchView.include({
+        
+        // Prevent focus of search field on mobile devices
+        toggle_visibility: function (is_visible) {
+             $('div.oe_searchview_input').last()
+                .one('focus', $.proxy(this.preventMobileFocus, this));
+            return this._super(is_visible);
+        },
+        
+        // It prevents focusing of search el on mobile
+        preventMobileFocus: function(event) {
+            if (this.isMobile()) {
+                event.preventDefault();
+            }
+        },
+        
+        // For lack of Modernizr, TouchEvent will do
+        isMobile: function () {
+            try{
+                document.createEvent('TouchEvent');
+                return true;
+            } catch (ex) {
+                return false;
+            }
+        },
+        
+    });
+    
+    var AppDrawer = Class.extend({
+        
+        LEFT: 'left',
+        RIGHT: 'right',
+        UP: 'up',
+        DOWN: 'down',
+        
+        isOpen: false,
+        $selectedAppLink: $(false),
+        keyBuffer: '',
+        keyBufferTime: 500,
+        keyBufferTimeoutEvent: false,
+        
+        init: function() {
+            this.$el = $('.drawer');
+            this.$el.drawer();
+            this.directionCodes = {
+                37: this.LEFT,  // Left arrow
+                39: this.RIGHT,  // Right arrow
+                38: this.UP,  // Up arrow
+                33: this.UP,  // page up
+                40: this.DOWN,  // Down arrow
+                34: this.DOWN,  // page down
+            };
```

Comment:
Maybe you can reuse the built-in [`$.hotkeys.specialKeys`](https://github.com/odoo/odoo/blob/9.0/addons/web/static/lib/jquery.hotkeys/jquery.hotkeys.js#L18-L26).